pkg/bootstrap/platform/azure.go

```diff
@@ -0,0 +1,202 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package platform
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+
+	"istio.io/pkg/log"
+)
+
+const (
+	AzureMetadataEndpoint  = "http://169.254.169.254"
+	AzureInstanceURL       = AzureMetadataEndpoint + "/metadata/instance"
+	AzureDefaultAPIVersion = "2019-08-15"
+	SysVendorPath          = "/sys/class/dmi/id/sys_vendor"
+	MicrosoftIdentifier    = "Microsoft Corporation"
+
+	AzureName     = "azure_name"
```

Comment:
can you add some comments for these.. what is name here?
Also the azure_ suffix - can you make it configurable by the caller as part of instantiating this env object? The caller may or may not want to prefix it depending upon the situation. And some fields may already have this prefix - which will make it double prefixed.